pkg/daemon/controller/controller.go

```diff
@@ -252,19 +270,50 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 
 	// Reconcile state of Windows services with the ConfigMap data
 	if err = sc.reconcileServices(cmData.Services); err != nil {
+		if failErr := sc.handleReconcileFailure(&node); failErr != nil {
+			klog.Errorf("error handling reconcile error, node may not be cordoned appropriately: %s", failErr)
+		}
 		return ctrl.Result{}, err
 	}
 
 	if err = sc.waitUntilNodeReady(); err != nil {
+		if failErr := sc.handleReconcileFailure(&node); failErr != nil {
+			klog.Errorf("error handling reconcile error, node may not be cordoned appropriately: %s", failErr)
+		}
 		return ctrl.Result{}, fmt.Errorf("error waiting for node to become ready")
 	}
-	// Version annotation is the indicator that the node was fully configured by this version of the services ConfigMap
-	if err = metadata.ApplyVersionAnnotation(sc.ctx, sc.client, node, desiredVersion); err != nil {
-		return ctrl.Result{}, fmt.Errorf("error updating version annotation on node %s: %w", sc.nodeName, err)
+	if err = sc.markNodeConfigured(&node, desiredVersion); err != nil {
+		return ctrl.Result{}, err
 	}
 	return ctrl.Result{}, nil
 }
 
+// handleReconcileFailure should be ran when a reconciliation fails due to issues on the Windows instance
```

Comment:
I don't understand why we need this function? We should cordon the node at the beginning of configuration and only uncordon if it is has been configured fully. I don't understand why cordon only if we fail more than 5 times.